src/commands/paranet/start-paranet-sync-commands.js

```diff
@@ -0,0 +1,112 @@
+import { ERROR_TYPE, PARANET_SYNC_FREQUENCY_MILLS } from '../../constants/constants.js';
+
+class StartParanetSyncCommands extends Command {
+    constructor(ctx) {
+        super(ctx);
+        this.commandExecutor = ctx.commandExecutor;
+        this.ualService = ctx.ualService;
+        this.blockchainModuleManager = ctx.blockchainModuleManager;
+        this.repositoryModuleManager = ctx.repositoryModuleManager;
+
+        this.errorType = ERROR_TYPE.PARANET.START_PARANET_SYNC_ERROR;
+    }
+
+    async execute() {
+        const operationId = this.operationIdService.generateId();
+
+        this.logger.info(
+            `Paranet sync: Starting Paranet sync command for operation id: ${operationId}`,
+        );
+
+        await this.commandExecutor.delete('paranetSyncCommand');
+
+        const promises = [];
+        this.config.assetSync?.syncParanets.forEach(async (paranetId) => {
+            if (!this.ualService.isUal(paranetId)) {
+                this.logger.info(`Paranet sync: paranetId ${paranetId} is not a valid UAL!`);
+                return Command.empty();
+            }
+
+            const contractKaCount = await this.blockchainModuleManager.getKnowledgeAssetsCount(
+                paranetId,
+            );
+            const cachedKaCount =
+                (await this.repositoryModuleManager.getParanetById(paranetId)?.kaCount) ?? 0;
+
+            if (cachedKaCount === contractKaCount) {
+                this.logger.info(
+                    `Paranet sync: KA count from contract and in DB is the same, nothing to sync!`,
+                );
+                return Command.empty();
+            }
+
+            this.logger.info(
+                `Paranet sync: Syncing ${contractKaCount - cachedKaCount + 1} assets...`,
+            );
+
+            const kaToUpdate = [];
+            for (let i = cachedKaCount; i <= contractKaCount; i += 50) {
+                const nextKaArray = this.blockchainModuleManager.getKnowledgeAssetsWithPagination(
+                    paranetId,
+                    i,
+                    50,
+                );
+                if (!nextKaArray.length) break;
+                kaToUpdate.push(...nextKaArray);
+            }
+
+            kaToUpdate
+                .map((ka) => ka.tokenId)
+                .forEach((tokenId) => {
+                    const commandData = {
+                        paranetId,
+                        operationId,
+                        tokenId,
+                    };
+
+                    promises.append(
+                        this.commandExecutor.add({
+                            name: 'paranetSyncCommand',
+                            data: commandData,
+                            period: PARANET_SYNC_FREQUENCY_MILLS,
+                        }),
+                    );
+                });
+
+            await this.repositoryModuleManager.updateParanetKaCount(paranetId, contractKaCount);
+        });
+
+        await Promise.all(promises);
```

Comment:
What will happen here if one of the promises fails, we will update Paranet ka count in database but not schedule all commands, so we will miss some token ids